### PR TITLE
Script for generating main libfile

### PR DESCRIPTION
**Pull request by cmurray-astro**
This is a work in progress! Fun with HDF5 formatting made this take longer than I thought, but figured I'd get what I have up ASAP. @galaxyumi plz comment! So far it only generates filters for WFC3 UVIS and IR, still looking to see if there is an easier way to grab all the filter options from stsynphot...

### CODE REVIEW

**Comment by cmurray-astro** [comment]
aaand tests failing because stsynphot is not found, and I suppose they'll always fail because in order to run stsynphot you need to download ~1 GB of supporting files...

**Comment by karllark** [comment]
> aaand tests failing because stsynphot is not found, and I suppose they'll always fail because in order to run stsynphot you need to download ~1 GB of supporting files...

Tests are failing because stsynphot is not a dependency.  This should just be added to the setup.cfg file.  And will not require the supporting files.  Just that python can load the module.  Can provide more details on the needed update to setup.cfg if it is not clear.

File: beast/tools/make_libfile.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        pwaves.append(newfilt.lpivot.value)
+        comments.append("")
+
+    # smash the contents arrays together
+    contents = np.array(
+        list(
+            zip(
+                tablenames,
+                observatories,
+                filternames,
```
Comment (galaxyumi) [inline beast/tools/make_libfile.py:201]: I think this should be instruments

File: beast/tools/make_libfile.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        # append dataset to the hdf5 filters group
+        f.create_dataset(filter_name, data=arr)
+
+        # generate filter instance to compute relevant info
+        newfilt = phot.Filter(wave, bp_avg, name=filter_name)
+
+        # populate contents lists with relevant information
+        tablenames.append(filter_name)
+        observatories.append("HST")
+        filternames.append(mode.upper)
```
Comment (galaxyumi) [inline beast/tools/make_libfile.py:151]: I think this should be instruments. Items could be WFC3, ACS, WFPC2, etc.

File: beast/tools/make_libfile.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    # Open hd5 file for writing
+    hf = h5py.File(__ROOT__ + "filters.hd5", "w")
+
+    # Create group for nice hierarchical structure
+    f = hf.create_group("filters")
+
+    # Define arrays for "contents" / descriptive information
+    tablenames = []
+    observatories = []
+    filternames = []
```
Comment (galaxyumi) [inline beast/tools/make_libfile.py:109]: filternames -> instruments

**Comment by galaxyumi** [comment]
> This is a work in progress! Fun with HDF5 formatting made this take longer than I thought, but figured I'd get what I have up ASAP. @galaxyumi plz comment! So far it only generates filters for WFC3 UVIS and IR, still looking to see if there is an easier way to grab all the filter options from stsynphot...

I left comment about the "INSTRUMENT" column in the content group. Otherwise, the script looks good to me. You are planning to add ACS and WFPC2 filters?

**Comment by cmurray-astro** [comment]
yes! although I will need help to figure out WFPC2... @karllark mentioned figuring out how to combine the 4 chips? if either of you have insight, please let me know!

**Comment by cmurray-astro** [comment]
Ok I added ACS_WFC, WFPC2 and GALEX filters. Please look through the list and verify that my curating is reasonable? It's very easy to include more! Currently, after discussion with Pey-Lian Lim, the best way to list the filters is via the rather unsatisfying custom list of strings...

**Comment by galaxyumi** [comment]
@cmurray-astro Here is the list of filters used in LUVIT in addition to WFC3.
ACS/WFC filters - F475W, F555W, F606W, and F814W
WFPC2 filters - F439W, F555W, F606W, and F814W

**Comment by galaxyumi** [comment]
> Ok I added ACS_WFC, WFPC2 and GALEX filters. Please look through the list and verify that my curating is reasonable? It's very easy to include more! Currently, after discussion with Pey-Lian Lim, the best way to list the filters is via the rather unsatisfying custom list of strings...

I see all the LUVIT filters, as well as the filters in my own HST programs, in the current list. Thanks Claire! As we briefly talked about the ionizing photon filters, I would like to keep them separate (at least for now). The way loading and applying them are different from other filters in the current BEAST codes because we compute the number of photons, not the fluxes in those bands.

**Comment by karllark** [comment]
Looks like a good list of filters to me.

**Comment by karllark** [comment]
Remember to update the title when this PR is ready for review/merging.

**Comment by cmurray-astro** [comment]
ready for review/merging!

**Comment by galaxyumi** [comment]
> 

Looks good to me!

**Comment by karllark** [comment]
Where can I get the new library file?  Need to replace the existing one.

**Comment by cmurray-astro** [comment]
https://www.dropbox.com/s/arx24r24vn2yfge/filters.hd5?dl=0

**Pull request merged**